admin-portal/src/app/_components/EventCard.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useEffect } from 'react';
+import axios, { AxiosResponse } from "axios";
+import EditCard from "./EditCard";
+
+interface EventCardProps {
+    id: string;
+    removeFromList: (id: string) => void;
+}
+
+// EventData interface
+interface ActivityContent {
+    [key: string]: unknown;
+}
+
+interface Activity {
+    type: string;
+    content: ActivityContent;
+    timeStart: Date;
+    timeEnd: Date;
+    active: boolean;
+}
+
+interface EventData {
+    _id: string;
+    name: string;
+    date: Date;
+    description: string;
+    location: {
+        type: string;
+        coordinates: number[];
+    };
+    organizerId: string;
+    tags: string[];
+    registeredUsers: string[];
+    activities: Activity[];
+}
+
+export default function EventCard({ id, removeFromList }: EventCardProps) {
+    const [showButtons, setShowButtons] = useState(false);
+    const [isEditing, setIsEditing] = useState(false);
+    const [eventData, setEventData] = useState<EventData>({
+        organizerId: "",
+        _id: "",
+        name: "",
+        date: new Date(),
+        description: "",
+        location: {
+            type: "",
+            coordinates: [],
+        },
+        tags: [],
+        registeredUsers: [],
+        activities: []
+    });
+
+    const deleteEvent = async () => {
+        try {
+            removeFromList(id);
+            const response: AxiosResponse = await axios.delete(`http://${process.env.IP_ADDRESS}:${process.env.PORT}/events/deleteEvent/${id}`);
+            return response;
+        } catch (err) {
+            console.log(err);
+            return err;
+        }
+    };
+    
+    const getEventById = async () => {
+        try {
+            const response: AxiosResponse = await axios.get(`http://${process.env.IP_ADDRESS}:${process.env.PORT}/events/getEventById/${id}`);
+            console.log("got event");
+            return response.data;
+        } catch (err) {
+            console.log(err);
+            return err;
+        }
+    };
+
+    useEffect(() => {
+        const fetchData = async () => {
+            const data = await getEventById();
+            setEventData(data);
+        };
+        fetchData();
+    }, []);
```

Comment:
As you can see in the useEffect here, you are updating the ENTIRE OBJECT instead of individual fields, thus making more sense to keep the state variable to be the entire object. Good stuff.